gtsfm/utils/prepare_nerfstudio.py

```diff
@@ -0,0 +1,159 @@
+"""Functions to create transforms.json file and resized images that are input to Nerfstudio
+
+Here I have modified code taken from Nerfstudio for parsing data in the Nerfstudio format.
+Original file at:
+https://github.com/nerfstudio-project/nerfstudio/blob/main/scripts/process_data.py.
+
+Author: Jon Womack
+"""
+import argparse
+import os
+import json
+import numpy as np
+from enum import Enum
+from pathlib import Path
+
+
+from gtsfm.utils import images, io
+import thirdparty.nerfstudio.colmap_utils as colmap_utils
+
+
+class CameraModel(Enum):
+    """Enum for camera types."""
+
+    OPENCV = "OPENCV"
+    OPENCV_FISHEYE = "OPENCV_FISHEYE"
+
+
+CAMERA_MODELS = {
+    "perspective": CameraModel.OPENCV,
+    "fisheye": CameraModel.OPENCV_FISHEYE,
+}
+
+
+def colmap_to_json(cameras_path: Path, images_path: Path, output_dir: Path, camera_model: CameraModel) -> int:
+    """Converts COLMAP's cameras.bin and images.bin to a JSON file.
+    Args:
+        cameras_path: Path to the cameras.txt file.
+        images_path: Path to the images.txt file.
+        output_dir: Path to the output directory.
+        camera_model: Camera model used.
+    Returns:
+        The number of registered images.
+    """
+
+    cameras = colmap_utils.read_cameras_text(cameras_path)
+    images = colmap_utils.read_images_text(images_path)
+
+    # Only supports one camera
+    camera_params = cameras[1].params
+
+    frames = []
+    for _, im_data in images.items():
+        rotation = colmap_utils.qvec2rotmat(im_data.qvec)
+        translation = im_data.tvec.reshape(3, 1)
+        w2c = np.concatenate([rotation, translation], 1)
+        w2c = np.concatenate([w2c, np.array([[0, 0, 0, 1]])], 0)
+        c2w = np.linalg.inv(w2c)
+        # Convert from COLMAP's camera coordinate system to ours
+        c2w[0:3, 1:3] *= -1
+        c2w = c2w[np.array([1, 0, 2, 3]), :]
+        c2w[2, :] *= -1
```

Comment:
I prefer you use GTSAM types and use `.inverse()` on those. Less things to test as we can be more confident in GTSAM's implementation.